tests/Traits/ComposerTestTrait.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Pantheon\Terminus\Tests\Traits;
+
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Trait ComposerTestTrait.
+ *
+ * @package Pantheon\Terminus\Tests\Traits
+ */
+trait ComposerTestTrait
+{
+    /**
+     * Run a composer command.
+     *
+     * @param string $command
+     *   The command to run.
+     * @param string|null $pipeInput
+     *   The pipe input.
+     *
+     * @return array
+     *   The execution's stdout [0], exit code [1] and stderr [2].
+     */
+    protected static function callComposer(string $command, ?string $pipeInput = null): array
+    {
+        $procCommand = sprintf('composer %s', $command);
+        if (null !== $pipeInput) {
+            $procCommand = sprintf('%s | %s', $pipeInput, $procCommand);
+        }
+
+        $process = proc_open(
+            $procCommand,
+            [
+                1 => ['pipe', 'w'],
+                2 => ['pipe', 'w'],
+            ],
+            $pipes,
+            dirname(__DIR__, 2)
+        );
+
+        if (!is_resource($process)) {
+            return ['', 1, sprintf('Failed executing command "%s"', $procCommand)];
+        }
+
+        $stdout = trim(stream_get_contents($pipes[1]));
+        fclose($pipes[1]);
+
+        $stderr = trim(stream_get_contents($pipes[2]));
+        fclose($pipes[2]);
+
+        $exitCode = proc_close($process);
+
+        return [$stdout, $exitCode, $stderr];
+    }
+
+    /**
+     * Run a composer command.
+     *
+     * @param string $command
+     *   The command to run.
+     * @param array $suffixParts
+     *   Additional command options added to the end of the command line.
+     * @param bool $assertExitCode
+     *   If set to TRUE, assert the exit code equals to zero.
+     */
+    protected function composer(string $command, array $suffixParts = [], bool $assertExitCode = true): ?string
+    {
+        if (count($suffixParts) > 0) {
+            $command = sprintf('%s %s', $command, implode(' ', $suffixParts));
+        }
+
+        [$output, $exitCode, $error] = static::callComposer($command);
+        if (true === $assertExitCode) {
+            $this->assertEquals(0, $exitCode, $error);
+        }
+
+        return $output;
+    }
+
+    /**
+     * Install Terminus 2 plugins.
+     *
+     * @param array $plugins
+     */
+    protected function installTerminus2Plugins(array $plugins = [])
+    {
+        $filesystem = new Filesystem();
+        $plugins2_dir = getenv('TERMINUS_PLUGINS2_DIR');
+        if (is_dir($plugins2_dir)) {
+            $filesystem->remove($plugins2_dir);
+        }
+        $filesystem->mkdir($plugins2_dir);
+        foreach ($plugins as $plugin) {
+            $this->composer(sprintf('create-project --no-dev -d %s %s', $plugins2_dir, $plugin));
```

Comment:
@uberhacker Looks like `::composer()` is required to install plugins only, consider using Symfony's Process component instead of `::composer()` and `::callComposer()`, ie:
```php
use Symfony\Component\Process\Process;

$process = new Process(sprintf('create-project --no-dev %s', $plugin), $plugins2_dir);
$process->mustRun();
```

Reply:
Interesting suggestion.  This would reduce code which I'm all in favor of doing.

Reply:
In test code it's also fine to just use `exec()` if that's simpler.